### DISCUSSION TITLE
Advice on select search before a search string is supplied

### DISCUSSION THREAD

**Discussion by judgej** [wireui/wireui#584] [Q&A]
When a value is selected in a synchronous select item, and you come to the form and tab into the select, it immediately drops down with some values. In the case of the country select below (which comes from an array at the back end, but it could just as easily come from the database) the first bunch of countries are provided.

![image](https://user-images.githubusercontent.com/395934/208641133-85c3d162-e5fd-4d5d-b4cd-65300648e53b.png)

Now, it makes sense if the selected item is in the list and shown, instead of off in a a further select. One approach could be for the front end to make a selection on just the single selected item, so a drop-down appears just with "United Kingdom" in rather than the "A" countries.

Another option could be to signal the back end to return the selected item and a couple before and after it. Not quite sure how that kind of selection could be achieved reliably.

![image](https://user-images.githubusercontent.com/395934/208642920-75ca8cd4-963b-4b27-979d-4d991e61f1fe.png)

Or it could be that I've set something up wrong or there is an option I could use but just don't understand it?

```html
                <x-select
                    label="Country of manufacture"
                    wire:model.debounce="country_of_manufacture_code"
                    placeholder="Start typing…"
                    :async-data="route('livewire-api.country-select')"
                    option-label="name"
                    option-value="id"
                />
```

I'm trying to find out just what teh front end sends to get that initial list of "A" countries, but my docker has stopped working, so I'll be back after a reboot...

**Comment by judgej** [comment]
When first gaining focus on a select, the front end sends a search of `null`. The back end API can do what it likes with that. No context is provided on whether somethings is already selected in the select (IMO it should send the current `selected` value(s) *as well as* the `search` null, but that would probably be a breaking change.

However, it is easy to add the current selected value yourself with the async data:

```html
                <x-select
                    label="Country of manufacture"
                    wire:model.debounce="country_of_manufacture_code"
                    placeholder="Start typing…"
                    :async-data="route('livewire-api.country-select', [
                        'currentSelection' => $this->country_of_manufacture_code,
                    ])"
                    option-label="name"
                    option-value="id"
                />
```

That way, if my country select API gets both a `search` of `null`, and a `currentSelection` of a country code, it can return a result that includes that country(s) in it, and the drop down options look a little more intuitive than Afganistan.

Or alternatively, my API could return a complely empty set when the search is `null`, so no countries show upo at all. That would be easiest, but gives less feedback to the user as a hint at what they have just tabbed into.

Anyway, interested to see how other people handle these things. My aim is ultimately to not throw a surprising or unexpected UI at the end user.

**Comment by judgej** [comment]
I think the problem may be partly in the example API controller. It will filter the records if either the `search` is truthy, or the `selected` is present. On focus, a non-truthy (`null`) `search` only is sent, and that neans *no* filtering, so everything is returned (with whatever limit in imposed).

**Comment by Uhasith** [comment]
@judgej I have facing some serious issues with these select fields. As a beginner, I'm running into some problems using this in my project. Could you take 5-10 minutes and check my problem. It would be a great help sir !

**Comment by Uhasith** [comment]
#714  @judgej This is my issue. I open a discussion for that. Could you have time take a look about my problem
